narwhals/_dask/namespace.py

```diff
@@ -135,6 +137,52 @@ def any_horizontal(self, *exprs: IntoDaskExpr) -> DaskExpr:
     def sum_horizontal(self, *exprs: IntoDaskExpr) -> DaskExpr:
         return reduce(lambda x, y: x + y, parse_into_exprs(*exprs, namespace=self))
 
+    def concat(
+        self,
+        items: Iterable[DaskLazyFrame],
+        *,
+        how: str = "vertical",
+    ) -> DaskLazyFrame:
+        import dask.dataframe as dd  # ignore-banned-import
+
+        if len(list(items)) == 0:
+            msg = "No items to concatenate"
+            raise ValueError(msg)
+        native_frames = [i._native_frame for i in items]
+        axis: int
+        join: str
+        if how == "vertical":
+            if not all(
+                tuple(i.columns) == tuple(native_frames[0].columns) for i in native_frames
+            ):
+                msg = "unable to vstack with non-matching columns"
+                raise TypeError(msg)
+            axis = 0
+            join = "inner"
+        elif how == "horizontal":
+            all_column_names: list[str] = list(chain(*[i.columns for i in native_frames]))
+            if len(all_column_names) != len(set(all_column_names)):
+                duplicates = [
+                    i for i in all_column_names if all_column_names.count(i) > 1
+                ]
+                msg = (
+                    f"Columns with name(s): {', '.join(duplicates)} "
+                    "have more than one occurrence"
+                )
+                raise TypeError(msg)
```

Comment:
Either this:
```suggestion
                raise AssertionError(msg)
```
and maybe we can introduce custom exception later on to emulate what polars does (`DuplicateError`)